wordpress-mobile/WordPress-Android
# Extract site content fragment

Fixes #13884

We've been discussing with @malinajirka that the need to react to site changes in the my site view model makes the code unnecessarily complicated. There is also a chance of wrong behaviour after you change the selected site because we always use and update the previous view model. In the end we've reached a conclusion that the best possible solution is to extract the items related to the selected site into a separate fragment with its own view model. This component is only shown when there is a selected site and the fragment is recreated when the site is updated. Let me know what you think of my changes! There should be no functionality change to the My Site logic but the result looks cleaner 👍 

To test:
- Turn on the My Site improvements flag and restart the app
- Check the My Site works as expected (I've smoke tested the functionality and it seems OK, the tests pass with some minor changes).

PR submission checklist:
